Bugfix/519 extra line break added to file link

mattcoleanderson opened the pull request:
closes #519 

@ShootingKing-AM I've tested the change manually, however I ran into issues running the python test scripts. So I wouldn't be surprised if the e2e test fails initially.

mattcoleanderson committed:
remove line break from file link

mattcoleanderson committed:
remove line break from file link in py script

mattcoleanderson commented:
### Before

<img width="297" alt="image" src="https://github.com/ObsidianToAnki/Obsidian_to_Anki/assets/49250378/f6c1b1f7-1626-4763-8048-5dce89e47099">

### After: 

<img width="256" alt="image" src="https://github.com/ObsidianToAnki/Obsidian_to_Anki/assets/49250378/2d8d3ee6-ec64-4b48-a253-ec02a4276148">

ShootingKing-AM commented:
Thanks for the PR. Can this be made as an setting, which is default Off. Since many people would be used to the line break by now.

mattcoleanderson commented:
> Thanks for the PR. Can this be made as an setting, which is default Off. Since many people would be used to the line break by now.

I can work on that.

mattcoleanderson committed:
dynamically add new setting fields to the settings tab

mattcoleanderson committed:
add toggle to settings for prepending file link line break

mattcoleanderson commented:
> Thanks for the PR. Can this be made as an setting, which is default Off. Since many people would be used to the line break by now.

@ShootingKing-AM I've added a toggle to the settings for prepending the line break.

I've also added some code to main.js that when loading settings, will merge the default settings with settings saved to data.json. I noticed that prior behavior lead to newly added default settings not being added to the settings menu unless you deleted you data.json. Now that's fixed

mattcoleanderson commented:
@ShootingKing-AM Any chance of merging this PR?

ShootingKing-AM commented:
@mattcoleanderson this is blocked by https://github.com/ObsidianToAnki/Obsidian_to_Anki/issues/447 :(

samawelch commented:
Thanks for fixing this, @mattcoleanderson. If it's not too much to ask, could you tell me how I'd overwrite the current plugin files (installed inside Obsidian) with the changes you've made?

mattcoleanderson commented:
I'm closing this since it looks as though this repo is no longer maintained
